File: Laba23/Factory.cpp
#include "Factory.h"
#include <stdexcept>
using std::logic_error;

Shape* Factory::createShape(Point* cords, int number)
{
	if (number == 3)
	{
		return new Triangle(cords);
	}
	else if (number == 4)
	{
		return new Rectangle(cords);
	}
	else
	{
		throw logic_error("WRONG_SHAPE");
	}
}

File: Laba5-6/Factory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Laba56
{
    class Factory
    {
        public Shape createShape(Point[] cords, int number )
        {
			if (number == 3)
			{
				return new Triangle(cords);
			}
			else if (number == 4)
			{
				return new Rectangle(cords);
			}
			else
			{
				throw new ArgumentOutOfRangeException("WRONG_SHAPE");
			}
		}
    }
}
